learn/work/fairness-detectors/index.qmd

```diff
@@ -0,0 +1,290 @@
+source(here::here("common.R"))
+```
+
+```{r}
+#| label: "load"
+#| include: false
+library(tidymodels)
+pkgs <- c("tidymodels")
+
+theme_set(theme_bw() + theme(legend.position = "top"))
+```
+
+
+The popularity and capability of language models like GPT-4, which generate human-like text based on simple prompts, has rapidly increased in recent years. The releases of more powerful next-generation models are often accompanied by the development of GPT *detectors*, which are machine learning models intended to estimate the probability that inputted text was generated by a GPT.
+
+In 2022, a group of researchers collected a corpus of essays, some of which were written by real humans and others that were generated using various language models. Notably, some of the essays from real humans were written by people who did not write in English natively, while others were written by people who did.
+
+::: callout-note
+What does it mean to be a "native writer" of a language? Keep an eye out for this idea later on.
+:::
+
+The researchers passed each essay to a number of marketed GPT detectors, recording the predicted probabilities that each essay was written by a human or GPT model. Juxtaposing detector predictions for papers written by native and non-native English writers, Liang et al. argued in their paper [*GPT detectors are biased against non-native English writers*](https://doi.org/10.1016/j.patter.2023.100779) that GPT detectors disproportionately classify real writing from non-native English writers as AI-generated. Some news media channels that cite the study use it as evidence that GPT detector models behave unfairly. The statements 1) *GPT detectors are biased against non-native English writers*, 2) *GPT detectors disproportionately classify real writing from non-native English writers as AI-generated*, and 3) *GPT detector models behave unfairly* are only equivalent, though, given a shared set of social values surrounding what *fairness* means.
+
+Perhaps the core question that machine learning fairness as a research field has tried to address is exactly what a machine learning model acting fairly entails. As a recent primer notes, "\[t\]he rapid growth of this new field has led to wildly inconsistent motivations, terminology, and notation, presenting a serious challenge for cataloging and comparing definitions" (Mitchell et al. 2021). Fairness is a social construct, and means different things to different people; the operationalization of fairness as a quantitative metric is not at all trivial. The aim of evaluating a machine learning system's fairness is thus to state clearly one's definition of fairness, critique what that definition means in the problem context, and assess fairness under the chosen definition rigorously. The tidymodels packages in R provide a suite of functionality to support this kind of analysis, and we'll use the data from Liang et al. as an example to demonstrate a fairness-oriented analysis of model performance.
```

Comment:
Is it possible to link to the papers at the bottom of the page using citations? Or not worth the trouble?
https://quarto.org/docs/authoring/footnotes-and-citations.html

Reply:
Thank you for the nudge—this was the right move. Done in [4bbd9f9](https://github.com/tidymodels/tidymodels.org/pull/48/commits/4bbd9f9f19be00d627ab6cfad3d2b2ef558b75e9). :)